c/meterpreter/workspace/dump_sam/dump_sam.vcxproj

```diff
@@ -112,10 +113,11 @@
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
       <BufferSecurityCheck>false</BufferSecurityCheck>
       <FunctionLevelLinking>false</FunctionLevelLinking>
-      <CallingConvention>StdCall</CallingConvention>
+      <CallingConvention>Cdecl</CallingConvention>

```

Comment:
I think `Cdecl` is actually the [default](https://learn.microsoft.com/en-us/cpp/cpp/cdecl?view=msvc-170) calling convention on Windows. I found that `StdCall` calling convention was set for `dump_sam` and `elevator` projects and it was not compatible with the assembly code responsible for the Trampoline logic in `ReflectiveDLLInjection`, which assume the stack is being cleaned up by the caller. I couldn't find a reason why these two projects were using `StdCall` and testing revealed that changing the calling convention to `Cdecl` was not breaking anything. I might be wrong, though.

Reply:
cdecl is the C standard.  That said, Win32 API uses stdcalll (https://learn.microsoft.com/en-us/cpp/cpp/stdcall?view=msvc-170)
Win64 API uses a modified fastcall, IIRC.  I want to say 4 parameters in the registers and the rest on the stack.  I cannot remember the epilogue of the top of my head, though.

Reply:
Yeah @bwatters-r7 is correct, CDECL is standard for Posix binaries IIRC. Windows only uses CDECL for functions with a variable number of arguments (e.g. `printf`) and uses stdcall for the rest in 32-bit mode. For 32-bit code the primary difference between the two is who is responsible for stack cleanup, in CDECL it's the caller because they know how many arguments they pushed and for STDCALL it's the callee because they know how many arguments they were expecting.

Reply:
These changes were necessary since `StdCall` breaks the logic in the assembly code. Note that it only applies for x86 and `Cdecl` is the compiler default for functions where the calling convention is not explicitly declared. `__vectorcall` is used by default on x64.

I have no idea why these two projects (`dump_sam` and `elevator`) were specifically changed to `StdCall`. I couldn’t find any reference in PRs and they work fine using the default `Cdecl` calling convention. Note that `metsrv` and all the other extensions use the default `Cdecl` on x86.

If I remember well, the issue with `StdCall` was with the arguments provided by `rdi*` functions. It is possible to update the assembly to make it work with this calling convention, but it will require two different codes, one for these two projects (`StdCall`) and another one for `metsrv` and the other extensions. I thought easier to just set `Cdecl` back.